x-pack/plugins/observability_solution/infra/public/pages/metrics/hosts/hooks/use_host_count.ts

```diff
@@ -32,24 +32,6 @@ export const useHostCount = () => {
       const filters: QueryDslQueryContainer = {
         bool: {
           ...query.bool,
```

Comment:
```suggestion
        bool: query.bool,
```

Reply:
We need to create a new `bool` filter object and combine `query.bool` with some basic filters. It seems that would quite work as expected

```
bool: query.bool,
filter: [
  ...query.bool.filter,
  {
    exists: {
      field: 'host.name',
    },
  },
  {
    range: {
      [dataView?.timeFieldName ?? '@timestamp']: {
        gte: searchCriteria.dateRange.from,
        lte: searchCriteria.dateRange.to,
      },
    },
  },
],
```